### PR TITLE
Add APIs of circle invitation

### PR DESCRIPTION
**Pull request by devformatters**

### CODE REVIEW

**Review by jeremyOk — APPROVED** [review]

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: db/migrations/20210315152127_create_circle_invitation_table.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    t.bigInteger('circle_id').unsigned().notNullable()
+    t.boolean('accepted').defaultTo(false)
+
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('sent_at').defaultTo(knex.fn.now())
+    t.timestamp('accepted_at')
+    t.bigInteger('coupon_id').unsigned().notNullable()
+
+    // index
+    t.index(['inviter', 'circle_id'])
```
[the remark this answers is not in the document]
Comment (devformatters) [reply db/migrations/20210315152127_create_circle_invitation_table.js:23]: Nice point.

File: src/mutations/circle/invite.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  // process invitations
+  const invitations = []
+  for (const invitee of invitees) {
+    const { id, email } = invitee
+    const userId = id ? fromGlobalId(id).id : null
+
+    let invitation = await atomService.findFirst({
+      table: 'circle_invitation',
+      where: { circleId: circle.id, email, userId },
```
[the remark this answers is not in the document]
Comment (devformatters) [reply src/mutations/circle/invite.ts]: Added. Thx.

File: src/mutations/circle/invite.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    throw new UserInputError('invitees are required')
+  }
+
+  if (!months.includes(freePeriod)) {
+    throw new UserInputError('free period is invalid')
+  }
+
+  const circle = await atomService.findUnique({
+    table: 'circle',
+    where: { id: circleId },
```
[the remark this answers is not in the document]
Comment (devformatters) [reply src/mutations/circle/invite.ts]: Fixed.

**Review by robertu7 — APPROVED** [review]
🥼

**Pull request merged**